website/static/js/logFeed.js

```diff
@@ -1,238 +1,245 @@
+        self.activityLogs = m.prop();
+        self.logRequestPending = m.prop();
+        self.limitLogs = options.limitLogs;
+        self.failed = false;
+        self.paginators = m.prop([]);
+        self.nextPage = m.prop();
+        self.prevPage = m.prop();
+        self.firstPage = m.prop();
+        self.lastPage = m.prop();
+        self.totalPages = m.prop();
+        self.currentPage = m.prop();
+        self.pageToGet = m.prop();
+
+        self.getLogs = function _getLogs (url) {
+            self.activityLogs([]); // Empty logs from other projects while load is happening;
+            function _processResults (result){
+                result.data.map(function(log){
+                    log.attributes.formattableDate = new $osf.FormattableDate(log.attributes.date);
+                });
+
+                self.activityLogs(result.data);  // Set activity log data
+                self.nextPage = result.links.next;
+                self.prevPage = result.links.prev;
+                self.firstPage = result.links.first;
+                self.lastPage = result.links.last;
+
+                var params = $osf.urlParams(url);
+                var page = params.page ? params.page : 1;
+                self.currentPage(parseInt(page));
+                self.totalPages = Math.ceil(result.links.meta.total / result.links.meta.per_page);
             }
-            return '';
-        });
-    },
-    //send request to get more logs when the more button is clicked
-    fetchResults: function(){
-        var self = this;
-        self.loading(true); // show loading indicator
-
-        return $.ajax({
-            type: 'get',
-            url: self.url,
-            data:{
-                page: self.pageToGet()
-            },
-            cache: false
-        }).done(function(response) {
-            // Initialize LogViewModel
-            var logModelObjects = createLogs(response.logs); // Array of Log model objects
-            self.logs.removeAll();
-            for (var i=0; i<logModelObjects.length; i++) {
-                self.logs.push(logModelObjects[i]);
+            self.logRequestPending(true);
+            var promise = m.request({method : 'GET', url : url, config : xhrconfig});
+            promise.then(
+                function(result) {
+                    _processResults(result);
+                    self.logRequestPending(false);
+                    return promise;
+                }, function(xhr, textStatus, error) {
+                    self.failed = true;
+                    var message = 'Error retrieving logs for ' + self.node.id;
+                    Raven.captureMessage(message, {extra: {url: url, textStatus: textStatus, error: error}});
+                }
+            );
+        };
+
+        self.getCurrentLogs = function _getCurrentLogs (node, page){
+            if(!self.logRequestPending()) {
+                var url = _buildLogUrl(node, page, self.limitLogs);
+                return self.getLogs(url);
```

Comment:
Is it possible for `url` to be `undefined` here? It looks like `_buildLogUrl` will return `undefined` for nodes that aren't retracted/withdrawn.